src/router/index.ts

```diff
@@ -55,4 +60,40 @@ const router = createRouter({
   },
 });
 
+router.beforeEach(async (to, from) => {
+  const guestRoutes = ['login', 'logout', 'register']
+  
+  if (guestRoutes.includes(to.name)) {
+    return true
+  }
+
+  const user = useUserStore()
+
+  if (!user.isLoggedIn()) {
+    return user.logout()
+  }
+
+  if (Object.keys(user.profile).length === 0) {
+    await user.fetchProfile()
+  }
+
+  if (user.profile.phone_number_verified_at === null) {
+    const settingStore = useSettingStore()
+    await settingStore.fetchSettings()
+    settingStore.isSettingOpen('whatsapp-verification-is-active-on-user-registration')
```

Comment:
setting open ise yönlendire yapması gerekmez mi?
`if(settingStore.isSettingOpen('whatsapp-verification-is-active-on-user-registration')) {
router.push({name: 'verfiy-number})
}`